homeassistant/components/modbus/__init__.py

```diff
@@ -242,12 +242,14 @@
 
 FAN_SCHEMA = BASE_SWITCH_SCHEMA.extend({})
 
-SENSOR_SCHEMA = BASE_STRUCT_SCHEMA.extend(
-    {
-        vol.Optional(CONF_DEVICE_CLASS): SENSOR_DEVICE_CLASSES_SCHEMA,
-        vol.Optional(CONF_REVERSE_ORDER): cv.boolean,
-        vol.Optional(CONF_UNIT_OF_MEASUREMENT): cv.string,
-    }
+SENSOR_SCHEMA = vol.All(
+    cv.deprecated(CONF_REVERSE_ORDER),
+    BASE_STRUCT_SCHEMA.extend(
+        {
+            vol.Optional(CONF_DEVICE_CLASS): SENSOR_DEVICE_CLASSES_SCHEMA,
```

Comment:
We still need to allow the deprecated key, otherwise the it will be invalidated.
```suggestion
            vol.Optional(CONF_DEVICE_CLASS): SENSOR_DEVICE_CLASSES_SCHEMA,
            vol.Optional(CONF_REVERSE_ORDER): cv.boolean,
```

Reply:
Done